### PR TITLE
K8s local deployment and documentation updated

### PR DESCRIPTION
**Pull request by Tansito**
### Summary

This PR includes some new features that will make the life easier for local development:
- A script is included to be able to create a k8s cluster in your local machine to test in a nearer production-like environment: 

```shell
./docs/deployment/custom_function/local_cluster/deploy.sh
```
- Updated the README to be nearer to the current status of the project
- Documentation updated with the new k8s solution as a way to test locally
- The documentation was updated to remove the references to the patterns to start using the new functions therminology

#### Important

It's in the README and in the documentation but you need to install Kind to make use of the script and test it.

### CODE REVIEW

File: docs/deployment/custom_function/local_cluster/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#! /bin/bash
```
Comment (Tansito) [inline docs/deployment/custom_function/local_cluster/deploy.sh:1]: This is the script to create locally the cluster

File: README.md [hunk 2]
```diff
@@ -63,43 +63,40 @@ For user convenience, this section assumes that users will deploy the infrastruc
...
+      pip install qiskit-serverless
+      ```
+   1. Clone the Qiskit Serverless repository
+      ```shell
+      git clone https://github.com/Qiskit/qiskit-serverless.git
+      ```
+   1. Run Qiskit Serverless infrastructure
+      Execute the script to setup the cluster
+      ```shell
+      ./docs/deployment/custom_function/local_cluster/deploy.sh
```
Comment (korgan00) [inline README.md:89]: I tried to change to the `local_cluster` directory and then execute it and it doesn't work. It should be executed from the root.

We can add a `tox` command or something similar to simplify it.

**Review by korgan00 — CHANGES_REQUESTED** [review]
LGTM. After taking a look with you `colima` needed extra resources. It would be nice to have a warning or disclaimer about it in the documentation.

File: README.md (hunk 2 above)
Comment (Tansito) [reply README.md]: Modification applied here: [ae4bb54](https://github.com/Qiskit/qiskit-serverless/pull/1548/commits/ae4bb549669d64236af9180fbb65454666160f30)

The command is:
```shell
tox -e cluster-deploy
```

How do you see the command? Another proposal for the naming? Also @paaragon if you have any other proposal 👍

**Comment by Tansito** [comment]
Applied the comments @korgan00 , let me know if you see something more 😄

**Review by paaragon — APPROVED** [review]
Works great to me

**Review by korgan00 — APPROVED** [review]
LGTM

**Pull request merged**